src/hooks/context/useResetContext.js

```diff
@@ -28,17 +28,14 @@ const useResetContext = () => {
   const setRefreshCount = useSetRecoilState(resetAtom);
   const refreshCount = useRecoilValue(resetAtom);
 
-  const {setProfileId} = useProfileHook();
+  const {resetAuth} = useUserAuthHook();
 
   const resetAllContext = () => {
     setRefreshCount(new Date().getTime());
     DiscoveryAction.resetAllDiscovery(discoveryDispatch);
     resetAllNews(newsDispatch);
     setMyProfileResetAll(profileDispatch);
-    setProfileId({
-      anonProfileId: null,
-      profileId: null
-    });
+    resetAuth();
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `resetAuth` function is called at the end of the `resetAllContext` function. This is a good practice as it ensures that all other reset operations are completed before resetting the authentication. However, please ensure that any potential errors in the preceding reset operations (like `DiscoveryAction.resetAllDiscovery`, `resetAllNews`, `setMyProfileResetAll`) do not prevent `resetAuth` from being executed. If these functions throw an error, `resetAuth` might not be called, leaving the user in an authenticated state when they should be logged out. Consider wrapping these calls in a try-catch block to handle any exceptions and ensure `resetAuth` is always called.
```diff
   const resetAllContext = () => {
     setRefreshCount(new Date().getTime());
     try {
         DiscoveryAction.resetAllDiscovery(discoveryDispatch);
         resetAllNews(newsDispatch);
         setMyProfileResetAll(profileDispatch);
     } catch (error) {
         console.error('Error resetting context: ', error);
     }
     resetAuth();
  };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->